### PR TITLE
Add ability to choose another provider and course

### PR DESCRIPTION
**Pull request by chubberlisk**
### Context

With the current design, if a candidate wanted to make a course choice which we don't currently support they are left "stranded" on their user journey.

### Changes proposed in this pull request

This PR adds the option for "another" choice when choosing a provider and course. When choosing this, they are provided with the page to apply on UCAS.

### Screenshots (Updated 15/11, 13:35)

image

image

image

### Guidance to review

Would like feedback from someone from the design team to verify that this is want we want.

### Link to Trello card

[364 - Make the provider dropdown a radio button, and allow candidates to choose "other" (which goes to UCAS)](https://trello.com/c/m8GDiRSz/364-make-the-provider-dropdown-a-radio-button-and-allow-candidates-to-choose-other-which-goes-to-ucas)

### CODE REVIEW

**Comment by tvararu** [comment]
Btw that branch name is LONGE

**Comment by chubberlisk** [comment]
> Btw that branch name is LONGE

I stole the Trello card name and didn't stop to think about it... 😅

File: app/views/candidate_interface/course_choices/options_for_course.html.erb [hunk 1]
```diff
@@ -6,11 +6,17 @@
     <%= form_with model: Course.new, url: candidate_interface_course_choices_course_path(provider_code: params[:provider_code]), builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
       <%= f.govuk_error_summary %>
 
-      <h1 class="govuk-heading-xl">
-        <%= t('page_titles.which_course') %>
-      </h1>
+      <%= f.govuk_radio_buttons_fieldset :code, legend: { size: 'xl', text: t('page_titles.which_course') } do %>
+        <div class="govuk-!-margin-top-6">
+          <% @courses.map do |course| %>
+            <%= f.govuk_radio_button :code, course.code, label: { text: "#{course.name} (#{course.code})", size: 'm' } %>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/course_choices/options_for_course.html.erb]: All radio options should look the same (normal weight, standard size). I think removing the `size` attribute should fix this.

File: app/views/candidate_interface/course_choices/options_for_provider.html.erb [hunk 2]
```diff
@@ -6,11 +6,17 @@
     <%= form_with model: Provider.new, url: candidate_interface_course_choices_provider_path, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f| %>
       <%= f.govuk_error_summary %>
 
-      <h1 class="govuk-heading-xl">
-        <%= t('page_titles.which_provider') %>
-      </h1>
+      <%= f.govuk_radio_buttons_fieldset :code, legend: { size: 'xl', text: t('page_titles.which_provider') } do %>
+        <div class="govuk-!-margin-top-6">
+          <% @providers.map do |provider| %>
+            <%= f.govuk_radio_button :code, provider.code, label: { text: "#{provider.name} (#{provider.code})", size: 'm' } %>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/course_choices/options_for_provider.html.erb]: All radio options should look the same (normal weight, standard size). I think removing the `size` attribute should fix this.

**Pull request merged**